src/launcher_gui.js

```diff
@@ -21,89 +26,123 @@ app.on('second-instance', () => {
 });
 
 app.prependListener('ready', () => {
-	const { wizard } = require('./launcher_wizard');
-
-	const display = electron.screen.getPrimaryDisplay();
-	const sWidth = display.workAreaSize.width;
-	const width = 800;
-	const height = process.platform === 'win32' ? 418 : 380 + 8;
-
-	let windowOpts = {
-		x: (sWidth - width) / 2,
-		// y: (sHeight - height) / 2,
-		y: 100,
-		width: width,
-		height: height,
-		show: false,
-		icon: `${__dirname}/renderer/spring-icon.png`,
-		webPreferences: {
-			nodeIntegration: true,
-			enableRemoteModule: true,
-		},
-	};
-	windowOpts.resizable = false;
-	Menu.setApplicationMenu(null);
-	mainWindow = new BrowserWindow(windowOpts);
-
-	mainWindow.loadFile(`${__dirname}/renderer/index.html`);
-	if (isDev) {
-		// mainWindow.webContents.openDevTools();
-	}
+	// TODO: is there a better way to get yargs' positional args ? Did not find
+	// it ...
+	if (argv._[0]) {
+		// If a replay file was passed as argument to spring-launcher, then we
+		// don't open the GUI, we just start the replay.
+
+		// TODO: when we support downloading assets (map/game/engine) we might
+		// need to show the GUI anyway, to provide feedback on the
+		// downloads/ask for user confirmation.
+		const { launcher } = require('./engine_launcher');
```

Comment:
i think it might be best to keep all require statements at top level, to fail fast (js doesn't exactly catch a lot of errors at write-time) and reduce any slowdown after load

Reply:
I put as much as  I could top level, this one triggers a circular dependency, that's why it's here

Reply:
Well that's one ugly design, but it's my design, so I guess you don't have to fix it :D